From 8088cae4581d002abc0cf54ab76f2f179a24ffb7 Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@amd.com>
Date: Wed, 12 Mar 2025 16:58:59 +0000
Subject: [PATCH 5/6] Move `blas_ex/common_gemm_ex3` to different target

`clients/common/blas_ex/common_gemm_ex3.cpp` pulls in cblas via
`clients/common/common_helpers.hpp` ->
`clients/common/testing_common.hpp` ->
`clients/include/testing_common.hpp` ->
`clients/include/cblas_interface.hpp` -> `cblas.h`

While the `rocblas_clients_testing_common` target depends on BLAS,
`rocblas_clients_common` does not. This therefore moves the source file
to the former target. Both libraries get linked into one common
executable in a later step.
---
 clients/common/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/clients/common/CMakeLists.txt b/clients/common/CMakeLists.txt
index 7fabacf5..7ab67d91 100644
--- a/clients/common/CMakeLists.txt
+++ b/clients/common/CMakeLists.txt
@@ -147,11 +147,11 @@ if( BUILD_WITH_TENSILE )
   )
 endif()
 
-add_library(rocblas_clients_common OBJECT ${rocblas_testing_common_tensile_source} ${rocblas_common_source})
+add_library(rocblas_clients_common OBJECT ${rocblas_common_source})
 
 rocblas_client_library_settings( rocblas_clients_common )
 
-add_library(rocblas_clients_testing_common OBJECT ${rocblas_testing_common_source})
+add_library(rocblas_clients_testing_common OBJECT ${rocblas_testing_common_tensile_source}  ${rocblas_testing_common_source})
 
 rocblas_client_library_settings( rocblas_clients_testing_common )
 
-- 
2.47.1.windows.2

